dartagnan/src/main/java/com/dat3m/dartagnan/program/event/lang/spirv/SpirvCmpXchg.java

```diff
@@ -0,0 +1,78 @@
+package com.dat3m.dartagnan.program.event.lang.spirv;
+
+import com.dat3m.dartagnan.expression.Expression;
+import com.dat3m.dartagnan.program.Register;
+import com.dat3m.dartagnan.program.event.EventVisitor;
+import com.dat3m.dartagnan.program.event.common.RMWCmpXchgBase;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import static com.dat3m.dartagnan.program.event.Tag.Spirv.*;
+
+public class SpirvCmpXchg extends RMWCmpXchgBase {
+
+    private static final List<String> moStrength = List.of(RELAXED, RELEASE, ACQUIRE, ACQ_REL, SEQ_CST);
+
+    private final String scope;
+    private final String eqMo;
+    private final Set<String> eqTags;
+
+    public SpirvCmpXchg(Register register, Expression address, Expression cmp, Expression value, String scope, Set<String> eqTags, Set<String> neqTags) {
```

Comment:
Yes, they can have completely different semantics according to the spirv standard, including different avvis bits (limited to the validation constraints checked here). Because dartagnan doesn't support different tag sets, VisitorSpirvVulkan will fail compilation if tag sets are not equal.

Reply:
We actually have a way of supporting different tags by using assumes. See e.g. [lkmm CAS](https://github.com/hernanponcedeleon/Dat3M/blob/8237954ad41afa8c18ce68c9cf0303dc68b000d8/dartagnan/src/main/java/com/dat3m/dartagnan/program/processing/compilation/VisitorLKMM.java#L57).

Reply:
I don't like that it generates incorrect control dependencies, so I will keep it not implemented in spirv for now.